velox/exec/RowContainer.cpp

```diff
@@ -1275,36 +1275,47 @@ RowComparator::RowComparator(
   }
 }
 
-bool RowComparator::operator()(const char* lhs, const char* rhs) {
+int32_t RowComparator::compare(const char* lhs, const char* rhs) {
   if (lhs == rhs) {
-    return false;
+    return 0;
   }
   for (auto& key : keyInfo_) {
     if (auto result = rowContainer_->compare(
             lhs,
             rhs,
             key.first,
             {key.second.isNullsFirst(), key.second.isAscending(), false})) {
-      return result < 0;
+      return result;
     }
   }
-  return false;
+  return 0;
 }
 
-bool RowComparator::operator()(
+bool RowComparator::operator()(const char* lhs, const char* rhs) {
+  return compare(lhs, rhs) <= 0;
```

Comment:
If `lhs = rhs`, then we `return 0 <= 0`, which is `true`. It is supposed to be `false`.
should be `compare(lhs, rhs) < 0;`

Reply:
Don't we have a test to catch this?

Reply:
See comment below.